projects/hslayers/src/components/add-data/url/add-data-url.service.ts

```diff
@@ -33,4 +35,48 @@ export class HsAddDataUrlService {
       );
     });
   }
+
+  /**
+   * Selects a service layer to be added (WMS | ArcGIS Map Server)
+   * @param services Layer group of a service to select a layer from
+   * @param layerToSelect Layer to be selected (checked = true)
+   * @param selector Layer selector. Differs in between different services
+   */
+  selectLayerByName(
+    layerToSelect: string,
+    services: any,
+    selector: 'Title' | 'Name'
+  ): void {
+    if (!layerToSelect) {
+      return;
+    }
+    for (const service of services) {
+      if (service.Layer) {
+        for (const layer of service.Layer) {
+          if (layer[selector] == layerToSelect) {
+            layer.checked = true;
+            setTimeout(() => {
```

Comment:
Extract to function maybe to remove the duplication.

Reply:
good point, thanks